### PR TITLE
[tests] split platform and backend tests out

### PR DESCRIPTION
**Pull request by Saviq**
This way we can avoid conflicts when merging into full.

### CODE REVIEW

**Comment by Saviq** [comment]
Another potential approach for the include paths change: http://paste.ubuntu.com/p/nTjy5N8m3d/ - not really sure which I like better…

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -204,11 +131,14 @@ set_target_properties(mock_process
...
   RUNTIME_OUTPUT_DIRECTORY_RELEASE "${CMAKE_BINARY_DIR}/bin/mocks")
 
-add_executable(apparmor_parser
-  mock_apparmor_parser.cpp)
+add_dependencies(multipass_tests mock_process)
+
+target_include_directories(multipass_tests
+  BEFORE
+    PRIVATE ${CMAKE_SOURCE_DIR}/src/platform/backends/shared/linux
+)
```
Comment (ricab) [inline tests/CMakeLists.txt]: Why is this needed?

Comment (Saviq) [reply tests/CMakeLists.txt]: This replaces `PLATFORM_INCLUDES` above. Otherwise it matters where I put `add_subdirectories`.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -204,11 +131,14 @@ set_target_properties(mock_process
...
+add_subdirectory(linux)
+add_subdirectory(qemu)
+include(libvirt/CMakeLists.txt) # with add_subdirectory it's not possible to target_link_libraries
```
[the remark this answers is not in the document]
Comment (Saviq) [reply tests/CMakeLists.txt]: Because you can only `target_link_libraries` targets built in the same directory. A CMake limitation (by design, I believe).

File: tests/linux/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+target_sources(multipass_tests
+  PRIVATE
+    ${CMAKE_CURRENT_LIST_DIR}/../libvirt/mock_libvirt.cpp
```
[the remark this answers is not in the document]
Comment (Saviq) [reply tests/linux/CMakeLists.txt]: I suppose, it is theoretically possible though, that `libvirt` and `qemu` could be included in the other platforms.

Alternative is to place `mock_libvirt.{h,cpp}` somewhere shared?

File: tests/linux/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    ${CMAKE_CURRENT_LIST_DIR}/test_apparmored_process.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/test_platform_linux.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/test_qemuimg_process_spec.cpp
+    ${CMAKE_CURRENT_LIST_DIR}/test_snap_utils.cpp
```
[the remark this answers is not in the document]
Comment (Saviq) [reply tests/linux/CMakeLists.txt]: With `include` for libvirt this would point to `tests/`, not `tests/libvirt`.

File: tests/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (ricab) [reply tests/CMakeLists.txt]: OK, I just tried it and I see what you mean. For the record, [this](https://paste.ubuntu.com/p/wmb3vGKTcx/) would work, but it would reduce the encapsulation you are going for here. Yeah fine, I guess this is not worth the trouble of declaring an `ADDITIONAL_LIBS` var and filling it in the subdirs...

File: tests/linux/CMakeLists.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ricab) [reply tests/linux/CMakeLists.txt]: Right, I originally wrote the comment there but then skipped it because I imagined that would be the case. But then I saw it again here. For these regular cases, this looks intriguing.

File: tests/linux/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (ricab) [reply tests/linux/CMakeLists.txt]: Hmm, ok, I'll let you choose the best then.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (ricab) [reply tests/CMakeLists.txt]: Ah, alright.

File: tests/linux/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (Saviq) [reply tests/linux/CMakeLists.txt]: I think it's fine like this, explicit (it actually results in the file being twice in the sources for `multipass_tests`, but then allows to actually comment out `libvirt`).

File: tests/linux/CMakeLists.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (Saviq) [reply tests/linux/CMakeLists.txt]: It's just a variable like any other, pointing to the directory where the `CMakeLists.txt` is. I'd leave it be.

[the remark this answers is not in the document]
Comment (ricab) [reply tests/linux/CMakeLists.txt]: For future reference, the var is needed because, in older versions of cmake (before 3.13 I believe), relative paths would be resolved against the directory where the target was added (the parent dir, in this case), while in newer versions they would be resolved relatively to the current source directory (the same as `CMAKE_CURRENT_LIST_DIR` here). This way, it works on both sides. See https://cmake.org/cmake/help/latest/policy/CMP0076.html#policy:CMP0076
